pkg/minikube/update/update.go

```diff
@@ -236,3 +288,51 @@ func downloadChecksum(v semver.Version, downloadBinary, downloadLinkFormat strin
 
 	return hex.DecodeString(strings.TrimSpace(string(b)))
 }
+
+func checksumFor(h crypto.Hash, payload []byte) ([]byte, error) {
+	if !h.Available() {
+		return nil, errors.New("requested hash function not available")
+	}
+	hash := h.New()
+	hash.Write(payload) // guaranteed not to error
+	return hash.Sum([]byte{}), nil
+}
+
+func extractAndReplace(goos, dir string, archiveBytes []byte) string {
+	downloadedArchive := filepath.Join(dir, "minishift.tgz")
+	ungzip := filepath.Join(dir, "minishift.gzip")
+	extract := filepath.Join(dir, "minishift-extract")
+	binary := filepath.Join(extract, "minishift")
+
+	// Write archive file to disk
+	err := ioutil.WriteFile(downloadedArchive, archiveBytes, 0644)
```

Comment:
It not wrong, just fishy. The code just writes the file to _minishift.tgz_ regardless. It then checks the OS and if on Windows just unzips it, regardless of the extension. 

So why not download and safe the archive as is, with the proper filename. Then you decide whether to use tar or zip based on the filename. 

Also, as has as I can tell, this method is supposed returns the path to the extracted minishift executable - binary := filepath.Join(extract, "minishift"). However, on Windows the binary is _minishift.exe_. I have not tried, but would be very surprised if this works. One needs to handle the download and naming explicitly.